OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnAttack.cs

```diff
@@ -0,0 +1,172 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public class GrantConditionOnAttackInfo : ITraitInfo
+	{
+		[GrantedConditionReference]
+		[Desc("The condition type to grant.")]
+		public readonly string Condition = null;
+
+		[WeaponReference]
```

Comment:
`WeaponReference` is incorrect here.  These are referencing armament names, not weapons.